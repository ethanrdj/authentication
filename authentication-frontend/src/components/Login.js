import React from "react";
import { Link } from "react-router-dom";
import Alert from "./Alert";
import PropTypes from "prop-types";

const Login = ({
  handleChange,
  handleSubmitLogin,
  alertMessage,
  setAlertMessage,
}) => {
  return (
    <div className="Login">
      <h1 className="header">Login</h1>
      <form className="form" onSubmit={handleSubmitLogin}>
        <label
          className="form-labels"
          htmlFor="login-email"
          data-testid="username-id"
        >
          Email or Username
        </label>
        <input
          className="form-input"
          id="login-email"
          type="text"
          name="value"
          required
          onChange={handleChange}
        />
        <label
          className="form-labels"
          htmlFor="login-password"
          data-testid="password-id"
        >
          Password
        </label>
        <input
          className="form-input"
          id="login-password"
          type="password"
          name="password"
          required
          onChange={handleChange}
        />
        {alertMessage && <Alert alertMessage={alertMessage} />}

        <button className="form-button" type="submit">
          Submit
        </button>

        <Link
          className="link-to"
          to="/register"
          onClick={() => setAlertMessage("")}
        >
          Not a member? Sign up here!
        </Link>
      </form>
    </div>
  );
};

Login.propTypes = {
  handleChange: PropTypes.func,
  handleSubmitLogin: PropTypes.func,
  alertMessage: PropTypes.string,
  setAlertMessage: PropTypes.func,
};

export default Login;
