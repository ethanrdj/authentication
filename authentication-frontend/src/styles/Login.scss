.login-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.login-input {
  width: 60%;
}
