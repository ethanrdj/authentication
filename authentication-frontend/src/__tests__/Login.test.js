import { render } from "@testing-library/react";
import Login from "../components/Login";
import { BrowserRouter, Route } from "react-router-dom";
jest.mock("../components/Alert", () => () => <div data-testid="alert-id" />);

describe("Login", () => {
  it("Renders component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Route
          render={() => (
            <Login
              handleChange={jest.fn()}
              handleSubmitLogin={jest.fn()}
              alertMessage="Test alert message"
              setAlertMessage={jest.fn()}
            />
          )}
        />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Renders elements on the page", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Route
          render={() => (
            <Login
              handleChange={jest.fn()}
              handleSubmitLogin={jest.fn()}
              alertMessage="Test alert message"
              setAlertMessage={jest.fn()}
            />
          )}
        />
      </BrowserRouter>
    );

    expect(getByTestId("username-id")).toHaveTextContent("Email or Username");
    expect(getByTestId("password-id")).toHaveTextContent("Password");
  });

  it("renders the alert message", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Login
          handleChange={jest.fn()}
          handleSubmitLogin={jest.fn()}
          alertMessage="Test alert message"
          setAlertMessage={jest.fn()}
        />
      </BrowserRouter>
    );

    expect(getByTestId("alert-id")).toBeInTheDocument();
  });
});
