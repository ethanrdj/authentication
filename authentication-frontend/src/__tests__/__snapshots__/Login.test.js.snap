// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders component 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <h1
      class="header"
    >
      Login
    </h1>
    <form
      class="form"
    >
      <label
        class="form-labels"
        data-testid="username-id"
        for="login-email"
      >
        Email or Username
      </label>
      <input
        class="form-input"
        id="login-email"
        name="value"
        required=""
        type="text"
      />
      <label
        class="form-labels"
        data-testid="password-id"
        for="login-password"
      >
        Password
      </label>
      <input
        class="form-input"
        id="login-password"
        name="password"
        required=""
        type="password"
      />
      <div
        data-testid="alert-id"
      />
      <button
        class="form-button"
        type="submit"
      >
        Submit
      </button>
      <a
        class="link-to"
        href="/register"
      >
        Not a member? Sign up here!
      </a>
    </form>
  </div>
</DocumentFragment>
`;
